### PR TITLE
add linters to go-amino

### PR DESCRIPTION
**Pull request by tac0turtle**
ref #258

### CODE REVIEW

File: json_test.go [hunk 1]
```diff
@@ -363,7 +363,7 @@ func TestJSONCodecRoundTrip(t *testing.T) {
 		}
 
 		// Now check that the input is exactly equal to the output
-		uBlob, err := cdc.MarshalJSON(tt.out)
+		uBlob, _ := cdc.MarshalJSON(tt.out)
```
Comment (melekes) [inline json_test.go]: can we check err here? `assert.NoError(t, err)`

File: binary-encode.go [hunk 2]
```diff
@@ -441,7 +441,7 @@ func encodeFieldNumberAndTyp3(w io.Writer, num uint32, typ Typ3) (err error) {
 	if (typ & 0xF8) != 0 {
 		panic(fmt.Sprintf("invalid Typ3 byte %v", typ))
 	}
-	if num < 0 || num > (1<<29-1) {
```
Comment (tac0turtle) [inline binary-encode.go:444]: `uint32` can not be less than 0

File: reflect_test.go [hunk 3]
```diff
@@ -347,10 +347,10 @@ func TestCodecBinaryStructFieldNilInterface(t *testing.T) {
 	cdc.RegisterConcrete((*tests.InterfaceFieldsStruct)(nil), "interfaceFields", nil)
 
 	i1 := &tests.InterfaceFieldsStruct{F1: new(tests.InterfaceFieldsStruct), F2: nil}
-	bz, err := cdc.MarshalBinaryLengthPrefixed(i1)
+	bz, _ := cdc.MarshalBinaryLengthPrefixed(i1)
```
Comment (melekes) [inline reflect_test.go]: can we check err here? `assert.NoError(t, err)`

File: decoder.go [hunk 4]
```diff
@@ -270,7 +271,7 @@ func decodeNanos(bz *[]byte, n *int) (int32, error) {
 			return 0, err
 		}
 		// Validation check.
-		if nsec < 0 || maxNanos < nsec {
```
Comment (tac0turtle) [inline decoder.go:273]: `uint` can not be less than 0

**Review by melekes — APPROVED** [review]

**Pull request merged**